bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -861,21 +868,28 @@ impl Connection {
         let handshake = unsafe {
             s2n_connection_get_handshake_type_name(self.connection.as_ptr()).into_result()?
         };
-        // The strings returned by s2n_connection_get_handshake_type_name
-        // are static and immutable after they are first calculated
-        static_const_str!(handshake)
+        // SAFETY: constructed strings have a null byte appended to them
```

Comment:
not: I like to put safety comments inside the unsafe block just so the stay "attached" after refactors